core/src/main/scala/ai/lum/odinson/lucene/index/IncrementalOdinsonIndex.scala

```diff
@@ -94,17 +97,20 @@ class IncrementalOdinsonIndex(
         f(closeableSearcherHolder.searcher)
       }
     } catch {
-      case _: Throwable => throw new RuntimeException("what is the best way to deal with this?")
-//      case e: Throwable => {
-//        e.printStackTrace()
-//        throw e
+      case e: Throwable => throw new RuntimeException(s"Error using searcher: ${e.getStackTrace}")
```

Comment:
We probably in general need to be more careful with error handling, especially as this moves past research into the product realm.  Scala isn't a big help in this area.  At some point we should look through all the exceptions and devise a more formal strategy.  For example, there might be an OdinsonException which might always be logged somewhere, maybe even to a database or issue tracking system.  The original stack trace, e.getStackTrace, would be a field rather than a string in the message, etc.  If odinson is hosted, we would want to analyze the log files periodically, etc.

Reply:
On second thought, why don't we just throw in e as the cause of the RuntimeException for now.  It can be inspected or logged more easily from there than from the string.

Reply:
Will do.  The old exception was too opaque when I was debugging. : )